src/main/play-doc/operation/DynamicProxyConfiguration.md

```diff
@@ -0,0 +1,627 @@
+
+backend visualizer_backend
+  mode http
+    {{#eachBackendServer}}
+  server {{serverName}} {{host}}:{{port}} maxconn 1024
+    {{/eachBackendServer}}
+  {{/ifAcl}}
+
+{{/eachAcls}}
+
+
+{{#haproxyConf bundles}}
+{{serviceFrontends}}
+{{#unless (serviceFrontends)}}
+frontend dummy
+  bind 127.0.0.1:65535
+{{/unless}}
+{{serviceBackends}}
+{{/haproxyConf}}
+```
+
+The default configuration above is written in [Handlebars](http://handlebarsjs.com/) template, and thus the template makes use of variables and helper functions made available to the template.
+
+### Dynamic proxy configuration DSL
+
+The variables and helper functions exposed to the template forms the DSL required to generate HAProxy configuration from the information available within ConductR cluster.
+
+The following variables are available to the template.
+
+* [bundles](#bundles)
+* [overrideConfigDir](#overrideConfigDir)
+
+The following helper functions are available to the template.
+
+* [eachAcls](#eachAcls)
+* [ifAcl](#ifAcl)
+* [eachBackendServer](#eachBackendServer)
+  * [serverName](#serverName)
+  * [host](#host)
+  * [port](#port)
+* [haproxyHost](#haproxyHost)
+* [serviceFrontends](#serviceFrontends)
+* [serviceBackends](#serviceBackends)
+
+
+##### haproxyHost
+
+Variable - String.
+
+Host address of the proxy of which the configuration will be generated for.
+
+Example:
+
+```
+10.0.2.223
+```
+
+
+##### bundles
+
+Variable - array of JSON object.
+
+List of bundles returned by the ConductR bundles endpoint. This list is in the form of an arrays where each element is a bundle represented by a JSON object.
+
+The cluster-wide information of what has been deployed and what is currently running is being stored in this variable.
+
+Refer to [Query bundle state](#Query-bundle-state) endpoint for an example of such JSON.
+
+##### overrideConfigDir
+
+Variable - optional, String
+
+Physical directory location on the file system where configuration overrides has been unpacked.
+
+This variable is optional - it is only defined when HAProxy custom template for ConductR HAProxy has been supplied.
+
+Example value of the variable:
+
+```
+/tmp/0d24d10cb0d1af9bf9f7e0bf81778a61-e357d11e8a38228880aa57d976d6f83b/e357d11e8a38228880aa57d976d6f83b576fd95e21dc99e8e4d02d4b227b4be3
+```
+
+In the example above the content of the ConductR HAProxy configuration override zip has been unpacked into the directory `/tmp/0d24d10cb0d1af9bf9f7e0bf81778a61-e357d11e8a38228880aa57d976d6f83b/e357d11e8a38228880aa57d976d6f83b576fd95e21dc99e8e4d02d4b227b4be3`.
+
+##### eachAcls
+
+Iterates through each request ACL exposed by the endpoints given a list of bundles.
+
+Used in conjunction with [ifAcl](#ifAcl) to provide customised configuration override given endpoint matches a particular `system`, `system version`, and `endpoint name`.
+
+Once iteration of all bundles have been completed, generates default HAProxy configuration for the bundles without configuration override.
+
+Input arguments:
```

Comment:
Make the sub headlines within these sections bold:

#### eachAcls
...

**Input arguments**
...

**Example**
...